### PR TITLE
expand cert rotation script

### PR DESCRIPTION
**Pull request by deads2k**
Add more steps for rotation

/assign @cgwalters

### CODE REVIEW

**Review by cgwalters — APPROVED** [review]
I gave this a quick skim, let's see what the rehearsals say!

**Comment by deads2k** [comment]
>+ oc config new-kubelet-bootstrap-kubeconfig
/bin/bash: line 34: bootstrap.kubeconfig: Permission denied

I wonder where there's scratch space for us to write.  @bear-redhat any advice?

File: ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh [hunk 1]
```diff
@@ -1,10 +1,56 @@
...
+# skip the AWS STS bound SA changes because we don't have the aws CLI or know where this info is stored.
+
+# generate new client certs for kcm and ks
+oc adm ocp-certificates regenerate-leaf -n openshift-config-managed secrets kube-controller-manager-client-cert-key kube-scheduler-client-cert-key
+
+# distribute trust across all known clients
+# update our local CA bundle so that when new serving certs are used for kube-apiserver we will trust them
+oc config refresh-ca-bundle
+# produce a new kubelet bootstrap kubeconfig (used to create the first CSR and establishes ca bundle)
+oc config new-kubelet-bootstrap-kubeconfig > bootstrap.kubeconfig
```
Comment (deads2k) [inline ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh:33]: it's this write here that is failing due to a lack of permissions .

Comment (cgwalters) [reply ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh:33]: Should work to use `/tmp`

**Comment by deads2k** [comment]
/pj-rehearse pull-ci-openshift-oc-master-e2e-aws-certrotation

**Comment by deads2k** [comment]
/pj-rehearse pull-ci-openshift-oc-master-e2e-aws-certrotation

**Comment by deads2k** [comment]
huh, infra failed


/pj-rehearse

**Comment by cgwalters** [comment]
Looking at this a bit, I see there's a `/tmp/shared/admin.kubeconfig` which is copied into `/tmp/admin.kubeconfig`.  The must-gather phase explicitly does
`export KUBECONFIG=/tmp/shared/admin.kubeconfig` though.  I guess in this case, we may actually want to overwrite `/tmp/shared/admin.kubeconfig` instead?

File: ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
+oc adm reboot-machine-config-pool mcp/worker mcp/master
+oc adm wait-for-node-reboot nodes --all
+
+# now that trust is distributed, create new certificates using new roots
+oc adm ocp-certificates regenerate-leaf -n openshift-kube-apiserver-operator secrets node-system-admin-client
+oc adm ocp-certificates regenerate-leaf -n openshift-kube-apiserver secrets check-endpoints-client-cert-key control-plane-node-admin-client-cert-key  external-loadbalancer-serving-certkey internal-loadbalancer-serving-certkey kubelet-client localhost-recovery-serving-certkey localhost-serving-cert-certkey service-network-serving-certkey
+oc adm wait-for-stable-cluster
+
+# create new admin.kubeconfig
+oc config new-admin-kubeconfig > /tmp/shared/admin.kubeconfig
```
Comment (deads2k) [inline ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh:49]: @cgwalters didn't appear to work.

Comment (wangke19) [reply ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh:49]: We copied the PR code in our CI job step without setting `shared` folder, it ran passed, see pr https://github.com/openshift/release/pull/40684/files#diff-96847061d6dc4135ed8ec2439e36e15b28bf83c11d9fdf5e143088a67e3bf7c5R78 and job logs:
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.11-vsphere-ipi-proxy-fips-cert-rotation-p2-f1/1674396986943475712/artifacts/vsphere-ipi-proxy-fips-cert-rotation-p2-f1/openshift-upgrade-qe-test-cert-rotation/build-log.txt

Comment (cgwalters) [reply ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh:49]: Hmmm.   I think this gets into a much larger confusion I have between the separation between "QE Prow jobs" and "non-QE Prow jobs" for lack of a better term...

Can I run these jobs on a PR?

But the reason things are failing here is there's something going on related to how the kubeconfig is being passed along in Prow that I don't think is being done in that QE flow.

Comment (wangke19) [reply ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh:49]: @cgwalters From QE side, we created and maintained a batch of test jobs for upgrade,  see https://github.com/openshift/release/tree/master/ci-operator/config/openshift/openshift-tests-private#trigger-a-new-test-for-an-existing-periodic-job-in-prow, I am not sure if  they could be executed on this PR, maybe the ENV is a little different. Will give it a try. 

The reason we copied this PR code into the PR 40684, in order to accelerate the testing of jobs in different upgrade paths and platforms.

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-stable-4.11-azure-ipi-fips-cert-rotation-p2-f14

File: ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh (hunk 2 above)
Comment (wangke19) [reply ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh:49]: @deads2k Try this,
```
oc config new-admin-kubeconfig > "${SHARED_DIR}/admin.kubeconfig"
oc --kubeconfig="${SHARED_DIR}/admin.kubeconfig" whoami
```

**Comment by wangke19** [comment]
Another thing, please update the following to ref file https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-ref.yaml,
```
ref:
  as: openshift-oc-certrotation
  from: cli
  grace_period: 5m
  timeout: 3h
  commands: openshift-oc-certrotation-commands.sh
```
We found that the execution time is longer than the default 2 hours on some platforms, that will cause the test to fail.

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.11-gcp-ipi-workload-identity-rt-fips-cert-rotation-p2-f1

**Pull request closed without merging**